src/core/ObservableQuery.ts

```diff
@@ -735,7 +735,11 @@ Did you mean to call refetch(variables) instead of refetch({ variables })?`);
       if (this.pollingInfo) {
         if (!isNetworkRequestInFlight(this.queryInfo.networkStatus)) {
           this.reobserve({
-            fetchPolicy: "network-only",
+            // Most fetchPolicy options don't make sense to use in a polling context, as
+            // users wouldn't want to be polling the cache directly. However, network-only and
+            // no-cache are both useful for when the user wants to control whether or not the
+            // polled results are written to the cache.
+            fetchPolicy: this.options.fetchPolicy === 'no-cache' ? 'no-cache' : 'network-only',
```

Comment:
This might need to be initialFetchPolicy
```suggestion
            fetchPolicy: this.options.initialFetchPolicy === 'no-cache' ? 'no-cache' : 'network-only',
```

Reply:
We're on 3.6.x, so maybe this changed, but this is the patch that's working for us there.

Reply:
Great point, thanks for the callout!